tool/sanity_check.dart

```diff
@@ -306,6 +308,69 @@ Future testPopup(
   await ensureBodyDoesNotContain(popupText);
 }
 
+Future testMenu(
+    Future<Null> ensureElementColor(WebElement element, String color),
+    Future<Null> ensureBodyContains(String text),
+    Future<Null> ensureBodyDoesNotContain(String text),
+    WebDriver driver) async {
+  print("Testing menu.");
+
+  final colorOption = "lightsalmon";
+  final subMenu = "Lights";
+  final colorInitial = "rgba(255, 0, 0, 1)";
+  final colorAfterClick = "rgba(255, 160, 122, 1)";
+
+  final colorText =
+      await driver.findElements(const By.className("menu-colorchanger")).first;
+  await ensureElementColor(colorText, colorInitial);
+
+  final menus =
+      await driver.findElements(const By.tagName("dropdown-menu")).toList();
+
+  await ensureBodyDoesNotContain(colorOption);
+  await ensureBodyDoesNotContain(subMenu);
+  menus.first.click();
+
+  // TODO(google) Remove sleep once PageObject testing classes are available
+  // Wait for the popup animation to complete
+  sleep(new Duration(milliseconds: 500));
+
+  await ensureBodyContains(subMenu);
+
+  var listItems = await driver
+      .findElements(const By.tagName("material-select-item"))
+      .toList();
+  for (var listItem in listItems) {
+    if ((await listItem.text).contains(subMenu)) {
+      await listItem.click();
+      break;
+    }
+  }
+
+  // TODO(google) Remove sleep once PageObject testing classes are available
+  // Wait for the popup animation to complete
+  sleep(new Duration(milliseconds: 250));
+
+  await ensureBodyContains(colorOption);
+
+  listItems = await driver
+      .findElements(const By.tagName("material-select-item"))
+      .toList();
+  for (var listItem in listItems) {
+    if ((await listItem.text).contains(colorOption)) {
+      await listItem.click();
+      break;
+    }
+  }
+  // TODO(google) Remove sleep once PageObject testing classes are available
+  // Wait for the popup animation to complete
+  sleep(new Duration(milliseconds: 250));
+
+  await ensureBodyDoesNotContain(colorOption);
+  await ensureBodyDoesNotContain(subMenu);
+  await ensureElementColor(colorText, colorAfterClick);
+}
+
```

Comment:
Maybe the sleep durations need to be tweaked? When I run the sanity tests I get:

```
> dart tool/sanity_check.dart
Waiting.......[Warning from Dart2JS on angular_components_example|web/main.dart]:
... [output trimmed] ...
[web] GET favicon.ico => Could not find asset angular_components_example|web/favicon.ico.
Page compiled and loaded.
Screenshot taken: screenshot.png
Testing app layout.
Testing button.
Testing tab.
Testing input.
Testing auto-suggest.
Testing dialog.
Testing menu.
Unhandled exception:
Body should contain 'Lights' at this point but doesn't.
#0      main.ensureBodyContains (file:///Users/chalin/git/dart/angular_components_example/tool/sanity_check.dart:46:9)
<asynchronous suspension>
#1      testMenu (file:///Users/chalin/git/dart/angular_components_example/tool/sanity_check.dart:338:27)
<asynchronous suspension>
#2      main (file:///Users/chalin/git/dart/angular_components_example/tool/sanity_check.dart:95:11)
<asynchronous suspension>
#3      _startIsolate.<anonymous closure> (dart:isolate-patch/dart:isolate/isolate_patch.dart:275)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:163)
```

Reply:
I've played around with the sleep durations, but that didn't have an effect.
It seems to be because the menu isn't visible, and so the animation doesn't happen. If I manually scroll just before the test expects to find "Lights" then the menu test passes.

Reply:
Oh, there is a missing await on the click.

Reply:
Good catch!